qlib/Mapper.qm

```diff
@@ -930,13 +931,42 @@ Mapper mapv(DataMap);
         }
 
         #! maps a single field to the target
-        private nothing mapFieldIntern(reference h, string key, hash rec, *int do_list) {
+        /**
+          * Performs the actual mapping
+          *
+          * @param h the hash to be updated with the mapped key/value pair
+          * @param key the column name (hash key) to be mapped (target field)
+          * @param rec input record - either single record of hash of lists (bulk);
+          *                           to increase performance, the input type
+          *                           (single record vs. bulk) is determined by
+          *                           the @ref do_list parameter. In case of
+          *                           bulk input, all the lists are supposed to
+          *                           have the same length @ref list_size
+          * @param do_list - whether the input record @ref rec is single record or
+          *                  bulk format (hash of lists)
+          * @param list_size - size of the lists in case the input is in bulk
+          *                    format (all lists must have the same length,
+          *                    asserted inside the function).
+          * @param crec - optional hash to be added to each processed record
```

Comment:
I think it is not that simple: 'rec' can be a bulk or not, and 'crec' can overwrite or add new keys in it. We probably can do some expanding of 'crec' and adding directly to 'rec' before calling `mapFieldIntern`, but it looked to me as too complex